### ISSUE TITLE
Add Access-Control-Allow-Origin to .well-known/matrix/client response

### ISSUE THREAD

**Issue by jo-so** [matrix-org/synapse#4593]
**Description:**

It would be helpful, if the response of *.well-known/matrix/client* contains the field `Access-Control-Allow-Origin: *`. Currently, I have to add this with Nginx to make the login of riot.im/staging work.

```
% nc -C ip6-localhost 8008 <<<$'GET /.well-known/matrix/client HTTP/1.0\n'
HTTP/1.0 200 OK
Content-Length: 55
Date: Fri, 08 Feb 2019 21:43:44 GMT
Content-Type: application/json
Server: Synapse/0.34.1.1

{"m.homeserver": {"base_url": "https://alea.gnuu.de/"}}
```

**Comment by turt2live** [comment]
Related: https://github.com/vector-im/riot-web/issues/7760

**Comment by n1trux** [comment]
This was apparently fixed in #4651 but if you get a 404 code the `Access-Control-Allow-Origin` header isn't set, which leads to other errors.

**Comment by richvdh** [comment]
fixed by #4651
